pandas/core/window/rolling.py

```diff
@@ -450,7 +457,36 @@ def homogeneous_func(values: np.ndarray):
             if values.size == 0:
                 return values.copy()
 
+            def _calculate_center_offset(window) -> int:
```

Comment:
None of this code below should be needed.

The `center` calculation should just adjust the indexers return in `calculate_variable_window_bounds` and everything else should work correctly.